presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewInformationExtractor.java

```diff
@@ -0,0 +1,149 @@
+import com.facebook.presto.sql.tree.SingleColumn;
+import com.facebook.presto.sql.tree.Table;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class MaterializedViewInformationExtractor
+        extends DefaultTraversalVisitor<Void, MaterializedViewInformationExtractor.MaterializedViewInfo>
+{
+    @Override
+    protected Void visitQuerySpecification(QuerySpecification node, MaterializedViewInfo materializedViewInfo)
+    {
+        super.visitQuerySpecification(node, materializedViewInfo);
+        if (node.getLimit().isPresent()) {
+            materializedViewInfo.setLimitClausePresented(true);
+        }
+        return null;
+    }
+
+    @Override
+    protected Void visitRelation(Relation node, MaterializedViewInfo materializedViewInfo)
+    {
+        // Currently it only handles Table instance cases (Do not handle Join, AliasedRelation, etc)
+        if (node instanceof Table) {
+            materializedViewInfo.setMaterializedViewDefinitionRelation(Optional.of(node));
+        }
+        return null;
+    }
+
+    @Override
+    protected Void visitSingleColumn(SingleColumn node, MaterializedViewInfo materializedViewInfo)
+    {
+        Expression baseColumnName = node.getExpression();
+        Optional<Identifier> baseColumnAlias = node.getAlias();
+        Identifier viewDerivedColumnName = baseColumnAlias.orElse(new Identifier(baseColumnName.toString()));
+
+        materializedViewInfo.putInBaseToViewColumnMap(baseColumnName, viewDerivedColumnName);
+        return null;
+    }
+
+    @Override
+    protected Void visitGroupBy(GroupBy node, MaterializedViewInfo materializedViewInfo)
+    {
+        materializedViewInfo.setMaterializedViewDefinitionGroupBy(Optional.of(ImmutableSet.copyOf(node.getGroupingElements())));
+        return null;
+    }
+
+    // Currently the following two methods will save the entire expression in the WHERE clause (logicalExpression or ComparisonExpression)
+    // TODO: Implement methods that save WHERE clause information in a better way which enable implication comparison (a's WHERE clause implies b's WHERE clause)
+    @Override
+    protected Void visitLogicalBinaryExpression(LogicalBinaryExpression node, MaterializedViewInfo materializedViewInfo)
+    {
+        materializedViewInfo.setMaterializedViewDefinitionWhere(Optional.of(node));
+        return null;
+    }
+
+    @Override
+    protected Void visitComparisonExpression(ComparisonExpression node, MaterializedViewInfo materializedViewInfo)
+    {
+        materializedViewInfo.setMaterializedViewDefinitionWhere(Optional.of(node));
+        return null;
+    }
+
+    public static final class MaterializedViewInfo
+    {
+        private final Map<Expression, Identifier> baseToViewColumnMap = new HashMap<>();
+        private Optional<Relation> materializedViewDefinitionRelation = Optional.empty();
```

Comment:
Lets remove materializedViewDefinition prefix from these variables? and call Relation to baseTable?